avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1445,7 +1446,7 @@ private Schema toConnectSchema(org.apache.avro.Schema schema, boolean forceOptio
       while (paramIt.hasNext()) {
         Map.Entry<String, JsonNode> field = paramIt.next();
         JsonNode jsonValue = field.getValue();
-        if (!jsonValue.isTextual()) {
+        if ((!field.getKey().endsWith(DOC_PROP) && !jsonValue.isTextual())) {
```

Comment:
Hmm, do we need the doc-specific thing? I think this might just be a more general oversight -- either we shouldn't even attempt to store the `null` parameters in the first place (which is probably the ideal, but we already have data now with `null`s in it) or we could just broaden this to support optional strings, i.e. reject it here `if (!jsonValue.isNull() && !jsonValue.isTextual())`